src/Services/ResultService.php

```diff
@@ -42,28 +50,76 @@ public function countQuestion(string $quizInstanceId)
 
     public function getQuestionsInstance($quizInstanceId)
     {
-        $textInstanceRepository = $this->repositoryManager->getRepository(QuestionInstanceRepository::class);
+        $textInstanceRepository = $this->repositoryManager->getRepository(QuestionInstance::class);
 
         return $textInstanceRepository->findBy(['quizInstanceId' => $quizInstanceId], [], 0, 0);
     }
 
     public function getTextInstance($id)
     {
-        $textInstanceRepository = $this->repositoryManager->getRepository(TextInstanceRepository::class);
+        $textInstanceRepository = $this->repositoryManager->getRepository(TextInstance::class);
 
-        return $textInstanceRepository->findBy(['questionInstanceId' => $id], [], 0, 0);
+        return $textInstanceRepository->findOneBy(['questionInstanceId' => $id]);
     }
 
     public function scoreResult(string $score, string $quizInstanceId): bool
     {
-        $quizInstanceRepository = $this->repositoryManager->getRepository(QuizInstanceRepository::class);
+        $quizInstanceRepository = $this->repositoryManager->getRepository(QuizInstance::class);
 
         try {
             $quizInstanceEntity = $quizInstanceRepository->find((int)$quizInstanceId);
             $quizInstanceEntity->setScore($score);
-            return $quizInstanceRepository->insertOnDuplicateKeyUpdate($quizInstanceEntity);
+            return $quizInstanceEntity->save($quizInstanceEntity);
         } catch (NoSuchRowException $e) {
             return false;
         }
     }
+
+    public function getQuestionsAnswersForQuizWithId(string $quizInstanceId): ?array
+    {
+        $questions = $this->getQuestionsForQuizId($quizInstanceId);
+
+        if ($questions === null) {
+            return null;
+        }
+
+        $questionDTOs = [];
+        foreach ($questions as $question) {
+            $answer = $this->getAnswerForQuestionId($question->getId());
+
+            if ($answer === null) {
+                return null;
+            }
+
+            $answer =
+                ($question->getType() === 'code')
+                ? $this->codeHighLighter->highlight($answer->getText())
+                : $answer->getText();
+
+            $questionDTOs[] = new QuestionDTO($question->getText(), $answer);
+        }
+
+        return $questionDTOs;
+    }
+
+    private function getQuestionsForQuizId(string $quizInstanceId): ?array
+    {
+        $questionInstanceRepository = $this->repositoryManager->getRepository(QuestionInstance::class);
+
+        try {
+            return $questionInstanceRepository->findBy(['quizInstanceId' => $quizInstanceId], [], 0, 0);
+        } catch (NoSuchRowException $e) {
+            return null;
+        }
```

Comment:
is this really necessary? can't you just return an empty array?

Reply:
This is for error handling and easy comparation in upper layers

Reply:
I would've went with the empty array.